repair/row_level.cc

```diff
@@ -65,11 +65,15 @@ struct shard_config {
     unsigned ignore_msb;
 };
 
-static bool inject_rpc_stream_error = false;
+namespace repair::row_level {
+    inline bool inject_rpc_stream_error = false;
```

Comment:
I'm afraid I didn't understand the motivation of these changes. I'll need a slightly more elaborate commit message to understand them.

Reply:
We can put  them under repair:: namespace instead of repair::row_level. I wanted to put everything for repair under repair namespace but did not do it because I wanted avoid cosmetic changes to make the backport harder. Since we are at the splitting, it is a good time to do so.  

However, I suggest not applying namespaces changes in this series.  We can do it in follow ups.

Reply:
Short answer: it's very unsafe to make put those symbols into global weak linkage. So I have to move them to the namespace to ensure safety.

I have updated patch description:

```
    repair: move global vars to namespace with weak ext linkage
    
    1. As we split repair/row_level.cc into a separate independent
    translation units we still need some shared objects that have package
    level visibility. In order to do that we need to change their linkage to
    global weak linkage.
    
    2. It's a bad practice to have global variables in global namespace.
    Especially, it is the case when they must have narrow visibility scope.
    Global names can also easilly collide or wrongly merge during linkage.
    
    3. For those specific names there are many places where are variables
    have the same name. Even though they all are kept in non global
    namespaces and benefit from local linkage it is dangerous that someone
    will make a mistake.
    
    4. writer, repair_meta and row_level require to share those.
    
    This patch makes those global variables safe to share in the package
    scope.
```

Reply:
@asias just a note. I have found that there are no places in the code that write `inject_rpc_stream_error`. is it intended? how you use it? in local debug build?